plugins/hls-fourmolu-plugin/src/Ide/Plugin/Fourmolu.hs

```diff
@@ -63,10 +64,26 @@ provider plId ideState typ contents fp fo = withIndefiniteProgress title Cancell
             . fmap (join . first (mkError . show))
             . try @IOException
             $ do
-                (exitCode, out, err) <-
+                CLIVersionInfo{noCabal} <- do -- check Fourmolu version so that we know which flags to use
+                    (exitCode, out, _err) <- readCreateProcessWithExitCode ( proc "fourmolu" ["-v"] ) ""
+                    let version = do
+                            guard $ exitCode == ExitSuccess
+                            "fourmolu" : v : _ <- pure $ T.words out
+                            pure $ T.splitOn "." v
+                    case version of
+                        Just v -> pure CLIVersionInfo
+                            { noCabal = v >= ["0", "7"]
+                            }
+                        Nothing -> do
+                            T.hPutStrLn stderr "couldn't get Fourmolu version"
```

Comment:
This is bad error reporting, but I'm about to revamp the logging for the whole plugin anyway, since I've been told off by @fendor for printing directly to `stdout`!

Reply:
https://github.com/haskell/haskell-language-server/pull/2959